File: 0695. Max Area of Island/Max Area of Island.py
class Solution:
    def deepFirstSearch(self, grid: List[List[int]], x: int, y: int) -> int:
        if x < 0 or x == len(grid) or y < 0 or y == len(grid[x]) or grid[x][y] == 0:
            return 0
        grid[x][y] = 0
        return 1 + self.deepFirstSearch(grid, x + 1, y) + self.deepFirstSearch(grid, x - 1, y) + \
                   self.deepFirstSearch(grid, x, y + 1) + self.deepFirstSearch(grid, x, y - 1)
    
    def maxAreaOfIsland(self, grid: List[List[int]]) -> int:
        maxArea = 0
        
        for i in range(len(grid)):
            for j in range(len(grid[i])):
                if grid[i][j] == 1:
                    maxArea = max(maxArea, self.deepFirstSearch(grid, i, j))
        
        return maxArea

File: 0695. Max Area of Island/Max Area of Island.java
class Solution {
    private int dfs(int[][] grid, int x, int y) {
        if (x < 0 || x == grid.length || y < 0 || y == grid[0].length || grid[x][y] == 0) return 0;
        grid[x][y] = 0;
        return 1 + dfs(grid, x + 1, y) + dfs(grid, x - 1, y) + dfs(grid, x, y + 1) + dfs(grid, x, y - 1);
    }
    
    public int maxAreaOfIsland(int[][] grid) {
        int res = 0;
        
        for (int i = 0; i < grid.length; i++) {
            for (int j = 0; j < grid[i].length; j++) {
                if (grid[i][j] == 1) {
                    res = Math.max(res, dfs(grid, i, j));
                }
            }
        }
        
        return res;
    }
}